gamepad.cpp

```diff
@@ -80,78 +86,197 @@ void Gamepad::handleInput()
         {
             if(event.isButton())
             {
-                switch (event.number)
-                {
-                    case EXIT_BUTTON:
-                        if(event.value == BUTTON_DOWN)
-                        {
-                            qInfo("Exit Pressed");
-                            exit(EXIT_BY_BUTTON);
-                        }
-                        break;
+                buttonState[ ButtonID (event.number) ] = bool (event.value);
```

Comment:
To się tyczy każdego castu, używaj tych ceplusplusowych: static_cast, const_cast, reinterpret_cast i dynamic_cast.